oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServerOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -245,13 +249,26 @@ public void setAccessTokenExpiresSkew(Duration accessTokenExpiresSkew) {
 	}
 
 	private Mono<OAuth2AuthorizedClient> refreshIfNecessary(ClientRequest request, ExchangeFunction next, OAuth2AuthorizedClient authorizedClient) {
-		if (shouldRefresh(authorizedClient)) {
+		ClientRegistration clientRegistration = authorizedClient.getClientRegistration();
+		if (AuthorizationGrantType.CLIENT_CREDENTIALS.equals(clientRegistration.getAuthorizationGrantType()) && hasExpired(authorizedClient)) {
```

Comment:
I wonder if a helper method would be better here?  Something like 

`isClientCredentialsGranType(authorizedClient)`

Reply:
Yep can do that

Reply:
Done